### PR TITLE
[DONE] Crafting UI duplicate item collapsing

### PR DESCRIPTION
**Pull request by acidia**
A QOL change for #23930, the introduction of a load of different bird eggs is a pain for users and is only going to get worse when you have 8 different bird eggs, 4 reptile eggs, 12 insect eggs, etc.
<a href="https://imgur.com/YZedjfJ"></a>
Left is before, middle and right is afterwards.  It just won't write identical entries twice.  Applies to tools and components but I think I'm the only one using identical names for nearly identical items at the moment.

<a href="https://imgur.com/Kaz6HIO"></a>
Still going to be able to pick whichever egg you want.

Super simple change but it isn't elegant, I tried creating a new cleaned list for the loop 
`        for( auto a = comp_list.begin(); a != comp_list.end(); ++a ) {`
but couldn't create a new "std::vector<<T>>" that would accept .push_back() using an auto variable.

### CODE REVIEW

**Comment by DracoGriffin** [comment]
Greatly looking forward to this change -- it will also be quite useful for modding.

Only one question/concern: is there a way to denote rotten items more clearly?

**Comment by acidia** [comment]
That is something entirely separate all together.  if it's something that is a pain I can append the rot status to the menu that pops up when you select which items you want to use.

File: src/requirements.cpp [hunk 1]
```diff
@@ -474,7 +474,20 @@ std::vector<std::string> requirement_data::get_folded_list( int width,
     for( const auto &comp_list : objs ) {
         const bool has_one = any_marked_available( comp_list );
         std::ostringstream buffer;
+        std::vector<std::string> buffer_has;
+        bool already_has;
         for( auto a = comp_list.begin(); a != comp_list.end(); ++a ) {
+            already_has = false;
+            for (auto cont : buffer_has){
```
Comment (ZhilkinSerg) [inline src/requirements.cpp]: `for( auto cont : buffer_has ) {`

File: src/requirements.cpp [hunk 2]
```diff
@@ -474,7 +474,20 @@ std::vector<std::string> requirement_data::get_folded_list( int width,
...
+                if (cont == a->to_string(batch) + a->get_color( has_one, crafting_inv, batch )){
+                    already_has = true;
+                    break;
+                }
+            }
+            if (already_has){
```
Comment (ZhilkinSerg) [inline src/requirements.cpp]: `if( already_has ) {`

File: src/requirements.cpp [hunk 3]
```diff
@@ -474,7 +474,20 @@ std::vector<std::string> requirement_data::get_folded_list( int width,
     for( const auto &comp_list : objs ) {
         const bool has_one = any_marked_available( comp_list );
         std::ostringstream buffer;
+        std::vector<std::string> buffer_has;
+        bool already_has;
         for( auto a = comp_list.begin(); a != comp_list.end(); ++a ) {
+            already_has = false;
+            for (auto cont : buffer_has){
+                if (cont == a->to_string(batch) + a->get_color( has_one, crafting_inv, batch )){
```
Comment (ZhilkinSerg) [inline src/requirements.cpp]: `if( cont == a->to_string( batch ) + a->get_color( has_one, crafting_inv, batch ) ) {`

**Comment by acidia** [comment]
I think I got them all

**Comment by DracoGriffin** [comment]
Got Travis to behave, and @ZhilkinSerg is your review satisfied?

**Pull request merged**